### PR TITLE
[BugFix] Fix query failed because of migration

### LINKED ISSUE #5160: query failed because of migration
### Steps to reproduce the behavior (Required)
1. start ingestion but do not write data
2. start migration
3. migration success
4. tablet and transaction gc in BE
5. prepare and commit and publish version
6. query the table

### Expected behavior (Required)

query success

### Real behavior (Required)

data lost in BE, and the query failed because of fail to prepare tablet reader with missed versions

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.1.3

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5160 

## Problem Summary(Required) ：

The reason for this problem is the concurrency control of ingesting and migration. 
There is a bug in migration check whether there is ingestion during migration.
And the data will be deleted wrong.

### CODE REVIEW

File: be/src/runtime/tablets_channel.cpp [hunk 1]
```diff
@@ -202,7 +202,8 @@ void TabletsChannel::add_chunk(brpc::Controller* cntl, const PTabletWriterAddChu
         std::lock_guard l1(_partitions_ids_lock);
         for (auto& [_, delta_writer] : _delta_writers) {
             if (UNLIKELY(_partition_ids.count(delta_writer->partition_id()) == 0)) {
-                delta_writer->abort();
+                // no data load, abort txn without printing log
```
[the remark this answers is not in the document]
Comment (ABingHuang) [reply be/src/runtime/tablets_channel.cpp:205]: there is a concurrent problem in #4333's scheme of remove the rollback log. This pr fix this bug.

**Comment by ABingHuang** [comment]
run starrocks_be_unittest

**Comment by ABingHuang** [comment]
run starrocks_be_unittest

File: be/src/storage/delta_writer.cpp [hunk 2]
```diff
@@ -244,11 +230,8 @@ Status DeltaWriter::close() {
                                                  _state_name(state)));
     case kClosed:
```
Comment (chaoyli) [inline be/src/storage/delta_writer.cpp:231]: 1. It seems the kClosed state is redundant.
2. And add a initialized -> committed state transition can avoid the rollback log.

Comment (ABingHuang) [reply be/src/storage/delta_writer.cpp:230]: 1. kClosed is used to flush last memtable, it is useful.
2. how?

**Review by chaoyli — DISMISSED** [review]

**Review by sevev — DISMISSED** [review]

File: be/src/storage/txn_manager.cpp [hunk 3]
```diff
@@ -372,6 +373,11 @@ Status TxnManager::rollback_txn(TPartitionId partition_id, TTransactionId transa
         LOG(INFO) << "rollback transaction from engine successfully."
```
Comment (meegoo) [inline be/src/storage/txn_manager.cpp:373]: log twice?

**Review by meegoo — CHANGES_REQUESTED** [review]

**Comment by ABingHuang** [comment]
run starrocks_be_unittest

**Review by chaoyli — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**